### ISSUE TITLE
D403 false positives for Napoleon-style properties

### ISSUE THREAD

**Issue by e-gebes** [PyCQA/pydocstyle#574]
Neither the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings) nor the [NumPy docstring Style guide](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) define how a property docstring shall look like. There are also no examples in their guides.

On the other hand, with [Napoleon](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html), a Sphinx extension for combining Google or NumPy style with reST markup in an easier readable way, the following syntax for properties is supported for both the Google and also the NumPy style (see also [here](https://github.com/sphinx-doc/sphinx/issues/2979#issuecomment-250549631) and [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html#example-numpy)):

```py
    @property
    def area(self):
        """float: The area of the square."""
        return self.length * self.length
```

The type description can also contain "of" and "or", like

    """list of str: ..."""
    """int or None: ..."""

That style triggers the warning:

    D403 First word of the first line should be properly capitalized

Please consider ignoring such type prefixes (everything before first colon?) **in properties for checking of D403**. Also for D401 and D405 this could make sense (for properties only): A type could theoretically be named "This" or could be named like a verb.
